### PR TITLE
Script for CLI database configuration

### PR DESCRIPTION
**Pull request by stefanoborini**
Introduces a script to perform changes to the database via CLI.
Built on top of another PR.

### CODE REVIEW

**Comment by kitchoi** [comment]
Would it be more appropriate to have it under `remoteappmanager/cli`, and make it one of the console_scripts entry_points in `setup.py`?  e.g. `remoteapp_dbutil = remoteappmanager.cli.dbutil:cli`?

**Comment by stefanoborini** [comment]
well, why not? seems definitely part of the suite. Give me 5 minutes.

File: doc/source/deployment.rst [hunk 1]
```diff
@@ -38,13 +38,21 @@ Deployment of the complete system in a single machine/VM.
 
 #. Generate the SSL certificates if you do not already have them. The
    resulting certificates will have names test.* because they are
-   self-signed and are not supposed to be used for production.  Once
-   created, copy them to the jupyterhub directory::
+   self-signed and are not supposed to be used for production.
```
Comment (kitchoi) [inline doc/source/deployment.rst]: As the paragraph is getting a bit long, could you bold this line and make sure people read it please?

File: doc/source/deployment.rst [hunk 2]
```diff
@@ -38,13 +38,21 @@ Deployment of the complete system in a single machine/VM.
...
+
+   Once created the certificates, copy them to the jupyterhub directory::
 
      cd ../scripts
      sh generate_certificate.sh
      cp test.* ../jupyterhub/
 
+#. Create the database. By default, this is a sqlite file::
+
+     python dbutil.py --db=~/remoteappmanager.db init
```
Comment (kitchoi) [inline doc/source/deployment.rst]: as the console script is available, this is cooler:
`remoteappdb --db=~/remoteappmanager.db init`

File: remoteappmanager/cli/remoteappdb/__main__.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+@click.argument("user")
+@click.argument("team")
+@click.pass_context
+def adduser(ctx, user, team):
+    """Add a user USER to a team TEAM."""
+    db = ctx.obj["db"]
+    session = db.create_session()
+    with orm.transaction(session):
+        orm_team = session.query(orm.Team).filter(
+            orm.Team.name == team).first()
```
Comment (kitchoi) [inline remoteappmanager/cli/remoteappdb/__main__.py]: `if orm_team is None: raise Exception`?

File: remoteappmanager/cli/remoteappdb/__main__.py [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        orm_user = session.query(orm.User).filter(
+            orm.User.name == user).first()
```
Comment (kitchoi) [inline remoteappmanager/cli/remoteappdb/__main__.py]: Similar to above.

File: remoteappmanager/cli/remoteappdb/__main__.py [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+@app.command()
+@click.argument("image")
+@click.argument("team")
+@click.option("--allow-home", type=click.BOOL, default=False)
+@click.option("--allow-team-view", type=click.BOOL, default=False)
+@click.option("--volume", type=click.STRING)
+@click.pass_context
+def expose(ctx, image, team, db, allow_home, allow_team_view, volume):
+    """Exposes a given application identified by IMAGE to a specific
+    team TEAM."""
```
Comment (kitchoi) [inline remoteappmanager/cli/remoteappdb/__main__.py]: The docstring might be too long.  The help shows this:
```
expose  Exposes a given application identified by...
```

Comment (kitchoi) [inline remoteappmanager/cli/remoteappdb/__main__.py]: Ah never mind this.

File: remoteappmanager/cli/remoteappdb/__main__.py [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+        for orm_app in session.query(orm.Application).all():
+            print("{}:{}".format(orm_app.id, orm_app.image))
+
+
+@app.command()
+@click.argument("image")
+@click.argument("team")
+@click.option("--allow-home", type=click.BOOL, default=False)
+@click.option("--allow-team-view", type=click.BOOL, default=False)
+@click.option("--volume", type=click.STRING)
```
Comment (kitchoi) [inline remoteappmanager/cli/remoteappdb/__main__.py]: e.g. `help="Application data volume, format={SOURCE}:{TARGET}:{MODE}"`

File: remoteappmanager/cli/remoteappdb/__main__.py [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+@click.pass_context
+def expose(ctx, image, team, db, allow_home, allow_team_view, volume):
```
Comment (kitchoi) [inline remoteappmanager/cli/remoteappdb/__main__.py]: `db` is taken from context, so it is not required here as a positional argument.

File: remoteappmanager/cli/remoteappdb/__main__.py [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+    with orm.transaction(session):
+        for orm_app in session.query(orm.Application).all():
+            print("{}:{}".format(orm_app.id, orm_app.image))
+
+
+@app.command()
+@click.argument("image")
+@click.argument("team")
+@click.option("--allow-home", type=click.BOOL, default=False)
+@click.option("--allow-team-view", type=click.BOOL, default=False)
```
Comment (kitchoi) [inline remoteappmanager/cli/remoteappdb/__main__.py]: minor suggestion: `is_flag=True`

**Comment by kitchoi** [comment]
`app expose` is the only command that I could not get it to work.

Other than that plus a couple of minor suggestions, the rest looks good to me.

**Comment by stefanoborini** [comment]
@kitchoi try again

**Comment by kitchoi** [comment]
:+1:

**Pull request merged**